### PR TITLE
[DM-24921] Updates based on review comments

### PR DESCRIPTION
**Pull request by rra**
- Discuss federated authentication issues
- Add a design for quotas
- Expand file storage requirements and note that the portal uses WebDAV
- Discuss token scoping and add a design for representing scope information
- Discuss initial user account creation and its constraints
- Mention a user administrative interface

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -86,10 +93,15 @@ Initial user authentication
...
 CILogon will be used as an OpenID Connect provider, so the output from that authentication process will be a JWT issued by CILogon and containing the user's identity information.
 
-The CILogon-provided identity will be mapped to a Science Platform user.
-Users will be able to associate multiple CILogon identities with the same Science Platform user.
+If the user has not previously been seen, they may or may not be able to automatically create a Science Platform account, depending on their institutional affiliation.
+Users affiliated with a US institution will be able to automatically create their account based on their affiliation as released by the InCommon federation via CILogon.
+Other users should be able to record enough information to allow the project to contact them (name and email, for example), and then be held for manual review and approval.
+The details of this approval flow will be fleshed out in later documents.
+
+Once the user's account has been craeted, the CILogon-provided identity will be mapped to a Science Platform account.
```
Comment (gpdf) [inline index.rst:101]: "created"

File: index.rst [hunk 2]
```diff
@@ -86,10 +93,15 @@ Initial user authentication
 Initial user authentication for most deployments will be done via CILogon using a web browser.
 CILogon will be used as an OpenID Connect provider, so the output from that authentication process will be a JWT issued by CILogon and containing the user's identity information.
 
-The CILogon-provided identity will be mapped to a Science Platform user.
-Users will be able to associate multiple CILogon identities with the same Science Platform user.
+If the user has not previously been seen, they may or may not be able to automatically create a Science Platform account, depending on their institutional affiliation.
```
Comment (gpdf) [inline index.rst:96]: I think it would be appropriate to footnote this discussion with a reference to LSE-279.

File: index.rst [hunk 3]
```diff
@@ -265,6 +306,39 @@ If the first point (direct use of JWTs by third-party services) becomes compelli
...
+The current design intention is to identify the use cases where a user would need a separate API token and provide a UI to create a token for that specific purpose, possibly with an advanced user escape hatch that would allow the user to make a more unrestricted choice of token scopes.
+This balance will be discussed in more detail in future documents as the requirements become clearer.
+
+Federated authentication
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The simplest design for API authentication, and indeed for the entire authentication system, is for all protected applications to run within the same Kubernetes cluster.
+Each separate Kubernetes cluster (the summit, for example) would have its own independent instantiation of this authentication system, without any cross-realm authentication.
+
+This may not be practical or desirable in practice.
```
Comment (gpdf) [inline index.rst:332]: It's worth saying in this context that different R/LSP instances (clusters?) may have different sets of authorized users.  Cross-cluster working capabilities would have to take this into account.

File: index.rst [hunk 4]
```diff
@@ -352,8 +434,8 @@ For example, Google Filestore (useful for prototyping and test installations) su
 Other possible file systems (such as cluster file systems like GPFS or Lustre) are generally not available as standard services in cloud environments, which are used for prototyping and testing and which ideally should match the Data Facility environment.
 
 AFS and related technologies such as AuriStor deserve some separate discussion.
-AFS-based file systems are uniquely able to expose the same file system to the user's local machine and to the notebook aspect and internal Science Platform services.
-This neatly solves the problem of synchronizing files from a user's machine to their running notebook or their collaborators, which would be a significant benefit.
+AFS-based file systems are uniquely able to expose the same file system to the user's local machine and to the notebook aspect, portal aspect, and internal Science Platform services.
```
Comment (gpdf) [inline index.rst:437]: This is not the only place it appears, but in general the project typography is to capitalize the Aspect names: "Notebook Aspect", "Portal Aspect", "[Web] API Aspect".

File: index.rst [hunk 5]
```diff
@@ -367,6 +449,9 @@ While having native file system support on the user's system would be extremely
 
 None of the other options seem sufficiently compelling over the availability and well-understood features of NFSv3.
 
+Remote access to storage
+^^^^^^^^^^^^^^^^^^^^^^^^
+
 This leaves the question of how to provide file system access from a user's local device.
```
Comment (gpdf) [inline index.rst:455]: Do you want to mention synchronization in addition to access?  The "I want to edit remotely" use case does basically presume an implementation that provides synchronization.  But maybe we don't have to / want to raise this detail at this point.

**Review by gpdf — COMMENTED** [review]
Looks good, thanks for taking all the previous comments into account!

After today's discussion, I wonder if it's worth footnoting that we are aware of CERNBox/OwnCloud as an existing example of a remote file access and synchronization service.

**Comment by rra** [comment]
Thanks! These should all now be addressed by https://github.com/lsst-sqre/sqr-039/pull/7.

**Pull request merged**